From b24cb32d0b7956186adf7288bcf61500e6620e94 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sun, 4 Mar 2012 11:57:39 +0100
Subject: [PATCH 17/35] ehci: drop old stuff

RH-Author: Hans de Goede <hdegoede@redhat.com>
Message-id: <1330862278-22314-3-git-send-email-hdegoede@redhat.com>
Patchwork-id: 37915
O-Subject: [PATCH 02/21] ehci: drop old stuff
Bugzilla: 758104
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Alon Levy <alevy@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

From: Gerd Hoffmann <kraxel@redhat.com>

Drop the "ehci under development" banner.
Drop unused & inactive (#if 0) code.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(Cherry picked from: 01cd4e98dc6eed96519f44f03de893364e0601ab)
---
 hw/usb-ehci.c |   30 +-----------------------------
 1 files changed, 1 insertions(+), 29 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/usb-ehci.c |   30 +-----------------------------
 1 files changed, 1 insertions(+), 29 deletions(-)

diff --git a/hw/usb-ehci.c b/hw/usb-ehci.c
index cb6ec2f..64967d7 100644
--- a/hw/usb-ehci.c
+++ b/hw/usb-ehci.c
@@ -1476,36 +1476,10 @@ static int ehci_process_itd(EHCIState *ehci,
                 }
             }
 
-#if 0
-            /*  In isoch, there is no facility to indicate a NAK so let's
-             *  instead just complete a zero-byte transaction.  Setting
-             *  DBERR seems too draconian.
-             */
-
-            if (ret == USB_RET_NAK) {
-                if (ehci->isoch_pause > 0) {
-                    DPRINTF("ISOCH: received a NAK but paused so returning\n");
-                    ehci->isoch_pause--;
-                    return 0;
-                } else if (ehci->isoch_pause == -1) {
-                    DPRINTF("ISOCH: recv NAK & isoch pause inactive, setting\n");
-                    // Pause frindex for up to 50 msec waiting for data from
-                    // remote
-                    ehci->isoch_pause = 50;
-                    return 0;
-                } else {
-                    DPRINTF("ISOCH: isoch pause timeout! return 0\n");
-                    ret = 0;
-                }
-            } else {
-                DPRINTF("ISOCH: received ACK, clearing pause\n");
-                ehci->isoch_pause = -1;
-            }
-#else
             if (ret == USB_RET_NAK) {
+                /* no data for us, so do a zero-length transfer */
                 ret = 0;
             }
-#endif
 
             if (ret >= 0) {
                 if (!dir) {
@@ -2376,8 +2350,6 @@ static int usb_ehci_initfn(PCIDevice *dev)
     pci_register_bar(&s->dev, 0, MMIO_SIZE, PCI_BASE_ADDRESS_SPACE_MEMORY,
                                                             ehci_map);
 
-    fprintf(stderr, "*** EHCI support is under development ***\n");
-
     return 0;
 }
 
-- 
1.7.7.6

